src/commands.ts

```diff
@@ -77,6 +78,7 @@ import { partition } from "./util";
 import * as nls from "vscode-nls";
 import typedConfig from "./config";
 import { toHgUri } from "./uri";
+import { LineChange, applyLineChanges } from "./diff";
```

Comment:
Doesn't vscode have a type of their own to represent Line Change?

Reply:
Please update the engine and the vscode types of you have to, we will have to do it anyway to implement the timeline view

Reply:
LineChange is in the [proposed API](https://github.com/microsoft/vscode/blob/ab7a5ec3ade55070225a70837e5c6c30322fd98c/src/vs/vscode.proposed.d.ts) file, not any specific engine version. The svn extension does it this way and it feels like a good compromise.

Reply:
Oh, ok then! 😄